src/test/java/com/google/step/datamanager/VoteManagerDatastoreTest.java

```diff
@@ -0,0 +1,70 @@
+package com.google.step.datamanager;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class VoteManagerDatastoreTest {
+
+  private static final long USER_ID_A = 123;
+  private static final long USER_ID_B = 456;
+  private static final long USER_ID_C = 789;
+
+  private static final long DEAL_ID_A = 111;
+  private static final long DEAL_ID_B = 222;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private final VoteManagerDatastore manager = new VoteManagerDatastore();
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testVote() {
```

Comment:
This test is actually testing a lot of different scenarios. Please split it up into the following:

- `getVotes_returnsZeroAtFirst()`
- `vote_incrementsVotesCorrectly()`
- `vote_doesNotIncrementForDuplicateVoteFromUser()`
- `vote_doesNotAffectOtherDeals()`

Reply:
Also, add additional tests to ensure the direction parameter is working properly. Perhaps:

* `vote_decrementsVotesCorrectly()`
* `vote_doesChangeVoteForSameUser()` (if the same user votes again with a different direction)